Notebooks/DC2Cookbook.ipynb

```diff
@@ -0,0 +1,559 @@
+    "\n",
+    "1. Build a master stack.  I suggest using [lsstsw](https://confluence.lsstcorp.org/display/LDMDG/The+LSST+Software+Build+Tool).\n",
+    "2. Set up the stack: e.g. `$> setup obs_lsstSim -t bNNNN`\n",
+    "3. Clone the package you want to patch on top of your stack `$> clone git@github.com:lsst/obs_lsstSim.git; cd obs_lsstSim`\n",
+    "4. Get the branch: `$> checkout tickets/DM-4302`\n",
+    "5. Set up just (-j) the cloned package (since the rest of the packages are already set up): `$> setup -j -r .`\n",
+    "6. Build the cloned package (this is necessary even for pure python packages): `$> scons opt=3`\n",
+    "7. Optionally install it in your stack: `$> scons install declare`\n",
+    "\n",
+    "This assumes the simulated images have landed in a directory called ```images```\n",
+    "in the current directory.  In the images directory, you'll need a ```_mapper``` file with contents\n",
+    "```python\n",
+    "lsst.obs.lsstSim.LsstSimMapper\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above file will tell the stack where to put the raw files and eimages.\n",
+    "\n",
+    "Setup the stack environment.  This will make the `LsstSimMapper` class available:\n",
+    "```bash\n",
+    "$> setup obs_lsstSim\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ingest the images from a directory called images to a repository called `input_data`.\n",
+    "There are some config overrides in the `ingest.py` file.\n",
+    "```bash\n",
+    "$> ingestImages.py images images/lsst_*.fits.gz --mode link --output input_data\n",
+    "```\n",
+    "Now you are setup to process the data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Process the image data using the DM stack\n",
+    "\n",
+    "Start here if you just want to exercise the DM stack.  If you didn't follow the steps above, first get the data and astrometry.net index files from\n",
+    "[here](https://lsst-web.ncsa.illinois.edu/~krughoff/data/gri_data.tar.gz).  Then untar the tarball in a working directory.\n",
+    "\n",
+    "After you have the data, you can start following the steps below to get, for example, forced photometry in three bands. First, set up the reference catalog for photometric and astrometric calibration:\n",
+    "```bash\n",
+    "$> setup -m none -r and_files astrometry_net_data\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create calibrated images from the input eimages.  This will write to a repository called output_data.  The --id argument defines the data to operate on.  In this case it means process all data (in this example the g, r, and i bands) with visit numbers between 840 and 879.  Missing data will be skipped.\n",
+    "```bash\n",
+    "$> processEimage.py input_data/ --id visit=840..879 --output output_data\n",
```

Comment:
One problem we ran into last time was that I specified data ids in the cookbook that weren't the same as the ones for production.  This was because I just wanted to reduce a manageable set and I copied the command lines into the cookbook verbatim.  It would be good to try to figure out how id can be specified for both a quick run and for a large scale production.